src/nisarqa/processing/insar/quiver_plots.py

```diff
@@ -382,4 +683,261 @@ def truncate_colormap(
+            # Since we do not have a DEM, as an approximation, assume that
+            # each target falls on the surface of the WGS 84 ellipsoid.
+            # (Ideally, we'd use the target height rather than assuming zero
+            # height. But we don't have access to this information in the GOFF
+            # input product, so a reasonable approx. is to use zero height.)
+            ellipsoid = isce3.core.WGS84_ELLIPSOID
+
+            # When we used `proj.inverse()` above, we did not care
+            # about height because the height was simply passed through.
+            # Here, for `ellipsoid.lon_lat_to_xyz()` we do care about height,
+            # but we do not have a DEM in GOFF nor GUNW products.
+            # So, assume each target falls on the WGS84 ellipsoid.
+            target_xyz = ellipsoid.lon_lat_to_xyz([lon, lat, 0])
+
+            # Run geo2rdr to get the target position in radar coordinates.
+            aztime, srange = isce3.geometry.geo2rdr_bracket(
+                xyz=target_xyz,
+                orbit=orbit,
+                # NISAR products are zero-Doppler, so construct a zero-Doppler LUT
+                doppler=isce3.core.LUT2d(),
+                wavelength=wavelength,
+                side=look_side,
+                **geo2rdr_params,
+            )
+
+            # 3) Use the az/rng offset values (in meters) at the coordinate
+            #    to offset ("shift") the point in the radar grid
+
+            # Azimuth Offsets are in meters, but azimuth on the radar grid is in
+            # seconds. So, convert `az_offsets_at_arrow_tails` from meters to
+            # seconds by using the ground track velocity (meters/second).
+
+            # Use the aztime of each target in radar coordinates to
+            # interpolate the orbit in order to get the
+            # platform position and velocity in ECEF coordinates.
+            platform_pos, platform_vel = orbit.interpolate(aztime)
+
+            grd_trk_vel_at_arrow_tail = nisarqa.get_ground_track_velocity(
+                target_xyz,  # ECEF position of target on the ground
+                platform_pos,  # az
+                platform_vel,
+            )
+
+            az_off_val_at_arrow_tail_in_seconds = (
+                az_offsets_at_arrow_tails[i, j] / grd_trk_vel_at_arrow_tail
+            )
+
+            # Shift the radar grid's azimuth values
+            aztime += az_off_val_at_arrow_tail_in_seconds
+
+            # Shift the radar grid's range values
+            # Note: range offset values were already in meters, so just add them
+            srange += rg_offsets_at_arrow_tails[i, j]
+
+            # 4) Run rdr2geo to convert back to (x_1, y_1) LLH
+
+            # Skip NaN pixels (this is usually geocoding fill)
+            if not np.isfinite(aztime) or not np.isfinite(srange):
+                x_tail_coords_shifted[i, j] = np.nan
+                y_tail_coords_shifted[i, j] = np.nan
+                continue
```

Comment:
This should happen immediately after computing `aztime` and `srange` using geo2rdr above. Let's move it up there.